### ISSUE TITLE
Token should be checked for encoded data

### ISSUE THREAD

**Issue by ex37** [SpoonX/sails-hook-authorization#25]
Hi! Nice hook.

https://github.com/SpoonX/sails-hook-authorization/blob/master/api/controllers/AuthController.js#L70

In case of passing the refresh token to `GET /auth/me` unexpected error is encountered. This is cause `req.access_token.user` doesn't exist in case of refresh token, though the token is completely valid. Probably policies should be updated, to do a check if it's the right access_token passed.

**Comment by RWOverdijk** [comment]
@ex37 You're right. Refresh tokens shouldn't be accepted as access tokens. Would you like to submit a PR for that? :)

**Comment by ex37** [comment]
@RWOverdijk done. Btw one question - what's the idea of having ensureToken policy? What is the possible use?

**Comment by VMBindraban** [comment]
I removed the `ensureToken` policy in our project.

**Comment by ex37** [comment]
well I got it
```
if (accessToken === null) {
	return next();
}
```
verifyToken.js policy is not forcing to have an access token, it just check if it is present, and if yes adds to req.access_token.

**Comment by RWOverdijk** [comment]
@ex37 I use it when I _need_ a token. Sometimes it's optional. Validate token only makes sure the token, if it exists, is valid. ensureToken forces you to have a token. Used together you can tweak behavior :)

**Comment by ex37** [comment]
@RWOverdijk yes it gives you more flexibility. But it's kind of dangerous, and cause anyway you can't use ensureToken without verifyToken, maybe it makes more sense to make ensureToken also verify token, so you can use it separately?

**Comment by RWOverdijk** [comment]
How is it dangerous? I disagree. As long as you use them together, they do exactly what they describe.

**Comment by ex37** [comment]
I meant it's dangerous in case you accidentally omit ensureToken. But yes, reading code once more time I can see if you use it without verify, it will just throw an error. So never mind.
